Keas.Mvc/Services/ReportService.cs

```diff
@@ -448,5 +454,136 @@ public async Task<IList<PeopleLeavingWithAssetsModel>> PeopleLeavingWithAssets(G
+        {
+            if (team == null)
+            {
+                team = await _context.Teams.SingleAsync(a => a.Slug == teamSlug);
+            }
+
+            if (expiresBefore == null)
+            {
+                expiresBefore = DateTime.Now.AddDays(30);
+            }
+
+            var userRoles = await _securityService.GetUserRoleNamesInTeamOrAdmin(team.Slug);
+
+            //TODO: Replace this method with the projections as we don't need all the data. Would need to modify the view too
+            var model = await ReportItemsViewModel.CreateExpiry(_context, expiresBefore.Value, team.Slug, showType, userRoles, _securityService);
+            return model;
+        }
+
+        public async Task<ReportItemsViewModel> ExpiringItems(Group group, DateTime? expiresBefore, string showType)
+        {
+            if (expiresBefore == null)
+            {
+                expiresBefore = DateTime.Now.AddDays(30);
+            }
+
+            //This is a group report, just give access to all
+            var model = new ReportItemsViewModel();
+            model.ItemList = new List<string>() { "All" };
+            model.ItemList.Add("Access");
+            model.ItemList.Add("Equipment");
+            model.ItemList.Add("Key");
+            model.ItemList.Add("Workstation");
+            model.ShowType = showType;
+            model.ExpiresBefore = expiresBefore.Value;
+
+            model.ExpiringItems = new List<ExpiringItemReportModel>();
+
+
+            var expiringAccess = await _context.AccessAssignments.Where(a => (showType == "All" || showType == "Access") &&
+                a.Access.Team.Groups.Any(g => g.GroupId == group.Id) && a.ExpiresAt <= expiresBefore)
+                .Include(a => a.Person).Include(a => a.Access).ThenInclude(a => a.Team).Select(ExpiringAccessProjection()).ToArrayAsync();
+
+            if (expiringAccess.Any())
+            {
+                model.ExpiringItems.AddRange(expiringAccess);
+            }
+
+            var expiringKey = await _context.KeySerials.Where(a => (showType == "All" || showType == "Key") &&
+                a.Key.Team.Groups.Any(g => g.GroupId == group.Id) &&
+                a.KeySerialAssignment.ExpiresAt <= expiresBefore)
+                .Include(a => a.Team).Include(k => k.KeySerialAssignment).ThenInclude(a => a.Person).Include(k => k.Key).Select(ExpiringKeysProjection()).ToArrayAsync();
+
+            if (expiringKey.Any())
+            {
+                model.ExpiringItems.AddRange(expiringKey);
+            }
+
+            var expiringEquipment = await _context.Equipment.Where(a => (showType == "All" || showType == "Equipment") &&
+                a.Team.Groups.Any(g => g.GroupId == group.Id) &&
+                a.Assignment.ExpiresAt <= expiresBefore)
+                .Include(a => a.Team).Include(e => e.Assignment).ThenInclude(a => a.Person).Select(ExpiringEquipmentProjection()).ToArrayAsync();

```

Comment:
What is the point of `.Include()` if you are just going to do a projection?  The projection selects out what it wants.  I'm curious to see the SQL here -- does it just ignore the include or does it grab of bunch of unnecessary "include" records and then throw them away?  Either way I think it can only hurt

Reply:
Ah. good point. I should have realized that.